pkg/kapp/preflight/registry_test.go

```diff
@@ -102,7 +109,186 @@ func TestRegistryRun(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			err := tc.registry.Run(nil, nil)
-			require.Equal(t, tc.shouldErr, err != nil)
+			require.Equalf(t, tc.shouldErr, err != nil, "Unexpected error %v", err)
+		})
+	}
+}
+
+func TestRegistryDirectConfig(t *testing.T) {
+	testCases := []struct {
+		name      string
+		registry  *Registry
+		config    map[string]any
+		shouldErr bool
+	}{
+		{
+			name: "preflight checks registered, no config",
+			registry: &Registry{
+				known: map[string]Check{
+					"disabledCheck": NewCheck(
+						nil,
+						func(cfg CheckConfig) error {
+							if cfg != nil {
+								return errors.New("config should not be present")
+							}
+							return nil
+						},
+						true,
+					),
+				},
+			},
+		},
+		{
+			name: "preflight checks registered, config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(
+						nil,
+						func(cfg CheckConfig) error {
+							if cfg == nil {
+								return errors.New("config should be present")
+							}
+							v, ok := cfg["foo"]
+							if !ok {
+								return errors.New("foo config not present")
+							}
+							if v != "bar" {
+								return errors.New("foo should equal 'bar'")
+							}
+							return nil
+						},
+						true,
+					),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+		{
+			name: "preflight checks registered, unexpected config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(
+						nil,
+						func(cfg CheckConfig) error {
+							if cfg == nil {
+								return errors.New("config should be present")
+							}
+							_, ok := cfg["foobar"]
+							if ok {
+								return errors.New("foo config should not present")
+							}
+							return nil
+						},
+						true,
+					),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			for _, check := range tc.registry.known {
+				err := check.SetConfig(tc.config)
+				require.Equalf(t, tc.shouldErr, err != nil, "Unexpected error %v", err)
+			}
+		})
```

Comment:
Why should we be concerned with testing the `SetConfig()` function on an individual `preflight.Check` implementation?

IMO we should _only_ be testing the `tc.registry.SetConfig()` function because that is actually what we are concerned with validating the functionality of, not the functionality of the individual checks that make up the registry. This does not preclude us from later writing tests for the individual preflight check implementations that we write, but as of now I don't see the value this particular set of tests is providing.

Reply:
To clarify why I don't see this particular set of tests providing value, I think really what we are doing here is testing the logic you've written in the individual test config functions (i.e https://github.com/everettraven/kapp/pull/2/files#diff-c55b40d43ffd8bf5ab782c17d750df0624f1f5c0558e03df4835ee16fa753971R174-R183) and not any of the logic we are attempting to test.

Reply:
I can see your point. This was written before I had figured out how to test `registry.SetConfig` method via `config.Conf`, which is more appropriate for this type of infrastructure.